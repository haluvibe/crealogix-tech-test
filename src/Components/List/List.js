import React from "react";
import PropTypes from "prop-types";
import uuid from "uuid";
import Header from "../Header/Header";

const List = ({ history }) => {
  return (
    <>
      <Header title="Counter History" />
      <ul>
        {history.map(number => (
          <li key={uuid.v4()}>{number}</li>
        ))}
      </ul>
    </>
  );
};

List.propTypes = {
  history: PropTypes.array.isRequired
};

export default List;
