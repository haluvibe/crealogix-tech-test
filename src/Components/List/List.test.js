import React from "react";
import { shallow } from "enzyme";
import List from "./List";

const history = [0, 1, 2];

const mockProps = {
  history
};

const mockUuidKey = "00000000-0000-0000-0000-000000000000";

jest.mock("uuid/v4", () => () => mockUuidKey);

const setUp = (props = mockProps) => {
  const wrapper = shallow(<List {...props} />);
  return wrapper;
};

describe("List", () => {
  let wrapper;
  beforeEach(() => {
    wrapper = setUp();
  });

  it("should render correctly with mock props", () => {
    expect(wrapper).toMatchSnapshot();
  });

  it("includes the Header component", () => {
    expect(wrapper.find("Header").props()).toEqual({
      title: "Counter History"
    });
  });

  it("renders list items mapped from props correctly", () => {
    const texts = wrapper.find("ul").map(node => node.text());
    expect(texts).toEqual(["012"]);
    expect(wrapper.find("ul").exists()).toEqual(true);
    expect(wrapper.find("li").length).toBe(3);
  });
});
