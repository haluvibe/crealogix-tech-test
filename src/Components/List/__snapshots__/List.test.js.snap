// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List should render correctly with mock props 1`] = `
<Fragment>
  <Header
    title="Counter History"
  />
  <ul>
    <li
      key="00000000-0000-0000-0000-000000000000"
    >
      0
    </li>
    <li
      key="00000000-0000-0000-0000-000000000000"
    >
      1
    </li>
    <li
      key="00000000-0000-0000-0000-000000000000"
    >
      2
    </li>
  </ul>
</Fragment>
`;
